src/index.ts

```diff
@@ -87,14 +87,22 @@ async function showShareDialog(sharingService: SharingService, notebookContent:
 async function handleNotebookSharing(
   notebookPanel: NotebookPanel | ViewOnlyNotebookPanel,
   sharingService: SharingService,
-  manual: boolean
+  manual: boolean,
+  onManualSave?: () => void
```

Comment:
Why optional?

My motivation when suggesting moving this to `handleNotebookSharing` was to ensure that if we call `handleNotebookSharing` elsewhere the logic would not be skipped. If this is marked as optional, then it can be skipped in new code.